### PR TITLE
sync groups fixture if user is removed from a group

### PR DESCRIPTION
**Pull request by snopoke**
See for context: http://manage.dimagi.com/default.asp?170165

### CODE REVIEW

File: corehq/apps/groups/models.py [hunk 1]
```diff
@@ -260,3 +286,40 @@ def save(self, *args, **kwargs):
...
+            removal = UserRemoval.objects.get(group_id=self.group_id, user_id=self.user_id)
+            removal.removed_on = self.removed_on
+            removal.save()
+        except UserRemoval.DoesNotExist:
+            self.save()
+
+    def __repr__(self):
+        return ("UserRemoval(group_id={self.group_id}, user_id={self.user_id}, "
+                "removed_on={self.removed_on})"
+        ).format(self=self)
```
Comment (dimagimon) [inline corehq/apps/groups/models.py]: E124 closing bracket does not match visual indentation

File: corehq/apps/groups/tests/test_groups.py [hunk 2]
```diff
@@ -1,30 +1,38 @@
 from couchdbkit import BadValueError
 from django.test import TestCase, SimpleTestCase
+from corehq.apps.groups.models import UserRemoval
 from corehq.apps.groups.models import Group
 from corehq.apps.users.models import CommCareUser
 
+DOMAIN= 'test-domain'
```
Comment (dimagimon) [inline corehq/apps/groups/tests/test_groups.py]: E225 missing whitespace around operator

File: corehq/apps/users/tests/group_fixture_tests.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+from django.test import TestCase
+from xml.etree import ElementTree
+from casexml.apps.case.tests.util import check_xml_line_by_line
+from corehq.apps.users.models import CommCareUser
+
+DOMAIN = 't1'
+
+FakeSyncLog = namedtuple('FakeSyncLog', 'date')
+
+class GroupFixtureTests(TestCase):
```
Comment (dimagimon) [inline corehq/apps/users/tests/group_fixture_tests.py]: E302 expected 2 blank lines, found 1

**Comment by snopoke** [comment]
@nickpell @czue what do you think of this approach?

**Comment by nickpell** [comment]
looks ok, although I don't follow how it fixes the original problem :confused:

**Comment by snopoke** [comment]
@nickpell decided to split it into 2 to make it clearer what's changing:
* https://github.com/dimagi/commcare-hq/pull/7147
* https://github.com/dimagi/commcare-hq/pull/7148

**Pull request closed without merging**